bindings/c/test/mako/mako.cpp

```diff
@@ -138,23 +133,66 @@ int cleanup(Database db, Arguments const& args) {
 				return -1;
 			}
 		}
-	}
-
-	// Delete tenants in batches after each keyspace is clean
-	if (args.total_tenants > 0) {
-		Transaction systemTx = db.createTransaction();
+	} else {
 		int batch_size = args.tenant_batch_size;
 		int batches = (args.total_tenants + batch_size - 1) / batch_size;
+		// First loop to clear all tenant key ranges
 		for (int batch = 0; batch < batches; ++batch) {
 			for (int i = batch * batch_size; i < args.total_tenants && i < (batch + 1) * batch_size; ++i) {
 				std::string tenant_name = "tenant" + std::to_string(i);
-				Tenant::deleteTenant(systemTx, toBytesRef(tenant_name));
+				// New transaction to get the tenant metadata
+				Transaction getTx = db.createTransaction();
+				while (true) {
+					auto result = Tenant::getTenant(getTx, toBytesRef(tenant_name));
+					const auto rc = waitAndHandleError(getTx, result, "GET_TENANT");
+					if (rc == FutureRC::OK) {
+						// Read the tenant metadata for the prefix and issue a range clear
+						if (result.get().has_value()) {
+							auto val = result.get().value();
+							const char* metadata = reinterpret_cast<const char*>(val.data());
+							rapidjson::Document doc;
+							doc.Parse(metadata);
+							if (!doc.HasParseError()) {
+								std::string tenantPrefix = (doc["prefix"].GetString());
+								tx.clearRange(toBytesRef(tenantPrefix + "/"), toBytesRef(tenantPrefix + "0"));
```

Comment:
Perhaps mako is only writing to the prefix that includes the forward slash, but to generalize and clear the whole range would need something like:

```suggestion
								tx.clearRange(toBytesRef(tenantPrefix), toBytesRef(strinc(tenantPrefix));
```